Git Formats/Git Diff.sublime-syntax

```diff
@@ -0,0 +1,567 @@
+
+    # mode <mode>..<mode>
+    # mode <mode>,<mode>..<mode> (for combined diff)
+    - match: ^mode(?= [106475,.]+)
+      scope: keyword.other.diff
+      push:
+        - include: pop-before-eol
+        - include: file-modes
+        - include: separator-twodot
+        # Combined mode can have multiple pre-commits
+        - include: separator-comma
+
+    # copy from <path>
+    # copy to <path>
+    # rename from <path>
+    # rename to <path>
+    - match: ^(?:copy|rename) (from|to)
+      push:
+        - include: pop-before-eol
+        - include: path-separators
+
+    # similarity index <number>
+    # dissimilarity index <number>
+    - match: ^(?:dis)?similarity index
+      push:
+        - include: pop-before-eol
+        - match: (100|[1-9]?[0-9])(%)
+          scope: meta.number.integer.decimal.git
+          captures:
+            1: constant.numeric.value.git
+            2: constant.numeric.suffix.git
+
+    # index <hash>..<hash> <mode>
+    - match: ^index(?= [0-9a-f,.]+)
+      scope: keyword.other.diff
+      push:
+        - include: pop-before-eol
+        - include: file-modes
+        - include: hash-empty-content
+        - include: hash
+        - include: separator-twodot
+        # Combined mode can have multiple pre-commits
+        - include: separator-comma
+
+  diff-content:
+    - meta_content_scope: meta.block.delta.diff
+    - include: pop-before-diff-header
+    - include: binary-deltas
+    - include: line-ranges-combined
+    - include: line-ranges-unified
+    - include: deltas-unified
+
+  deltas-unified:
+    - meta_prepend: true
+    - match: ^(?=-- {{eol}})
+      pop: 1
+
+  # https://github.com/git/git/commit/051308f6e9cebeb76b8fb4f52b7e9e7ce064445c
+  binary-deltas:
+    - match: ^GIT binary patch
+      scope: keyword.other.diff
+      push: binary-delta-content
+
+  binary-delta-content:
+    # Binary type
+    - match: ^(delta|literal) ([0-9]+)
+      captures:
+        1: support.function.diff
+        2: meta.number.integer.decimal.diff constant.numeric.value.diff
+
+    # Empty file
+    - match: ^(H)cmV\?d00001$
+      scope: constant.language.null.diff
+      captures:
+        1: punctuation.definition.string.begin.diff
+
+    # Base85-encoded binary data
+    - match: ^([A-Za-z]){{base85}}{1,66}$
+      scope: meta.string.diff string.other.raw.diff
+      captures:
+        1: punctuation.definition.string.begin.diff
+
+    # Bail if anything else is encountered
+    - match: ^(?!delta|literal|$)
+      pop: 2
+
+  # https://git-scm.com/docs/git-format-patch#Documentation/git-format-patch.txt---no-signatureltsignaturegt
+  signature:
+    # This is the Git version unless otherwise configured.  You will notice
+    # that once inside this context, you cannot leave.  That seems to be how
+    # it works for these files.
+    - match: ^(-- ){{eol}}
+      captures:
+        1: meta.separator.diff punctuation.section.block.diff
+      set:
+        - meta_content_scope: comment.block.diff
```

Comment:
Is it really worth a dedicated context or can this pattern just be added to `diffs`?

Seems rather unlikely to be subject of inheritance.

Instead probably use `signature` for signature content.

```yml
  diffs:
    - match: ^(diff) ((--)(?:git|combined|cc))
      captures:
        1: variable.function.diff
        2: variable.parameter.diff
        3: punctuation.definition.parameter.diff
      push: [diff-header, file-header-continue]
    # https://git-scm.com/docs/git-format-patch#Documentation/git-format-patch.txt---no-signatureltsignaturegt
    # This is the Git version unless otherwise configured.  You will notice
    # that once inside this context, you cannot leave.  That seems to be how
    # it works for these files.
    - match: ^(-- ){{eol}}
      captures:
        1: meta.separator.diff punctuation.section.block.diff
      set: signature

  signature:
    - meta_content_scope: comment.block.diff
 ```